sdccc/src/main/java/com/draeger/medical/sdccc/sdcri/testclient/TestClientImpl.java

```diff
@@ -354,17 +412,116 @@ void deviceFound(final ProbedDeviceFoundMessage message) {
         }
 
         sdcRemoteDevice.registerWatchdogObserver(this);
+        isConnected.set(true);
+    }
+
+    @Override
+    public synchronized CompletableFuture<Boolean> enableReconnect(final long timeoutInSeconds) {
+        final var actualTimeout = getReconnectFeatureTimeout(timeoutInSeconds);
+        LOG.debug("Reconnect enabled for {} seconds.", actualTimeout);
+
+        this.reconnectEnabled.set(true);
+        final var ctx = (LoggerContext) LogManager.getContext(this.getClass().getClassLoader(), false);
+        final var appender =
+                (TriggerOnErrorOrWorseLogAppender) ctx.getConfiguration().getAppender(APPENDER_NAME);
+        appender.setThreadNameWhitelist(buildThreadNameWhiteList());
+
+        this.providerWaitBarrier.reset();
+        this.reconnectFuture = new CompletableFuture<>();
+
+        final ScheduledExecutorService timeoutScheduler = Executors.newSingleThreadScheduledExecutor();
+        try {
+            reconnectScheduledTimeoutTask = timeoutScheduler.schedule(
+                    () -> {
+                        if (!reconnectFuture.isDone()) {
+                            if (isConnected.get()) {
```

Comment:
Due to "isConnected", on a reconnect, being reset prior to the "reconnectFuture" 's completion, this case may run on successful reconnect.

The scheduled task may need some proper synchronization with the other threads, but at least the comment at https://github.com/Draegerwerk/SDCcc/pull/216/files#diff-955d82d123e93ce4675e6524a4b941f007756b1589cfb87114b82fffa9e2d181R438 does not match what is actually happening.